services/system/AuthDelegate/src/AuthDelegate.cpp

```diff
@@ -40,6 +41,39 @@ namespace SystemService
       check(row.has_value(), "sender does not have an owning account");
    }
 
+   void AuthDelegate::stagedAccept(uint32_t staged_tx_id, psibase::AccountNumber actor)
+   {
+      check(getSender() == StagedTxService::service, "can only be called by staged-tx");
+
+      auto staged_tx = to<StagedTxService>().get_staged_tx(staged_tx_id);
+      auto actions   = staged_tx.action_list.actions;
+      check(actions.size() > 0, "staged tx has no actions");
+
+      auto row = db.open<AuthDelegateTable>().getIndex<0>().get(actions[0].sender);
+      check(row.has_value(), "staged sender does not have an owning account");
+      if (row->owner == actor)
```

Comment:
How is this going to work for multiple levels of indirection?

transact -> auth-delegate:root -> auth-delegate:prods.strong -> producers:prod1,prod2,...

Reply:
```
alice@stagedtx->propose(tokens@setcode->setCode); // staged tx 1
alice@stagedtx->propose(prods-strong@stagedtx->accept(1)); // staged tx 2
myprod@stagedtx->accept(2)
```

Reply:
Oh, I think I understand your question. Let me think about that

Reply:
Am I missing something? I think multiple levels of indirection works, but it's manual.